linker/Tests/Mono.Linker.Tests.Cases/Advanced/PreserveDependencyMethod.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.LinkerServices;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.Advanced {
+	class PreserveDependencyMethod {
+		public static void Main ()
+		{
+			B.Method ();
+			B.SameContext ();
+		}
+
+		class B
+		{
+			[Kept]
+			int field;
+
+			[Kept]
+			void Method2 (out sbyte arg)
+			{
+				arg = 1;
+			}
+
+			[Kept]
+			[PreserveDependency ("Mono.Linker.Tests.Cases.Advanced.C.Dependency1 ()")]
```

Comment:
Will strings without spaces work?

Ex: "Mono.Linker.Tests.Cases.Advanced.C.Dependency1()"

Ex: "Mono.Linker.Tests.Cases.Advanced.C.Dependency2`1(T[],System.Int32))"

Reply:
yes, tweaked the test